config/config.go

```diff
@@ -1044,7 +1044,16 @@ type Database struct {
 	Embedded DatabaseKine `json:"embedded,omitempty"`
 
 	// External defines that an external database should be used as the backend for the virtual cluster
-	External DatabaseKine `json:"external,omitempty"`
+	External ExternalDatabaseKine `json:"external,omitempty"`
+}
+
+type ExternalDatabaseKine struct {
+	DatabaseKine
+
+	// DBServerCredentials specifies database server connection information to be used by Platform to create a database
+	// and non-privileged user. A kine endpoint should be created using the database and user on Platform registration.
+	// This is optional.
+	DBServerCredentials string `json:"dbServerCredentials,omitempty"`
```

Comment:
I'm not sure if this really makes it clear what happens here. This is platform only functionality right? Also is this a reference to database credentials or the credentials themselves? I think we should rather name this something like `platformDatabaseCredentialsRef`.